### PR TITLE
GLTFExporter: Added ior and specular support

### LINKED ISSUE #25334: GLTFExporter is always exporting the iridescence extension and never IOR
### Description

Importing a valid glTF with the ior extension (renders just fine) and then exporting it results in a glTF without the ior extension and with the iridescence extension (??). It no longer renders fine when imported again:
image

This appears to be because 1) the exporter doesn't support the IOR extension like the importer does, and 2) the iridescence extension is apparently used for *all* `MeshPhysicalMaterial`s. I'll post a PR soon to fix these. Hoping I can sneak into r149. 

Edit: ah, I see now that the export extensions have not at all kept up with the import extensions. Okay, we'll see how much I can get done quickly...

### Reproduction steps

1. Go to https://modelviewer.dev/editor/
2. Select the Water Bottles
3. Note the conversion from Spec Gloss has worked (click the "Converted" button to see the extensions include specular and ior).
4. Click "Download Scene"
5. Open the zip, drop the GLB back into the editor.
6. Click "Info" and see the extensions now include only clearcoat and iridescence. 

### Code

```js
// code goes here
```

### Live example

[https://modelviewer.dev/editor/](https://modelviewer.dev/editor/)

### Screenshots

_No response_

### Version

r148

### Device

Desktop

### Browser

Chrome

### OS

MacOS

### PR DESCRIPTION
**Pull request by elalish**
Fixes: #25334

**Description**

Added glTF specular and ior extension export, and fixed clearcoat and iridescence so they only add those extensions if needed.

### CODE REVIEW

File: examples/jsm/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -53,6 +54,18 @@ class GLTFExporter {
 
 		} );
 
+    this.register( function ( writer ) {
```
Comment (gkjohnson) [inline examples/jsm/exporters/GLTFExporter.js:57]: nit: indentation doesn't look right on some of these lines.

File: examples/jsm/exporters/GLTFExporter.js [hunk 2]
```diff
@@ -2499,6 +2514,93 @@ class GLTFMaterialsVolumeExtension {
...
+
+		this.writer = writer;
+		this.name = 'KHR_materials_specular';
+
+	}
+
+	writeMaterial( material, materialDef ) {
+
+		if ( ! material.isMeshPhysicalMaterial || ( material.specularIntensity === 1.0 && 
+      material.specularColor.equals(DEFAULT_SPECULAR_COLOR) &&
```
Comment (mrdoob) [inline examples/jsm/exporters/GLTFExporter.js:2568]: Hmm, strange that the linter didn't notice this... @LeviPesin related to the latest CI changes maybe?

Comment (gkjohnson) [reply examples/jsm/exporters/GLTFExporter.js:2568]: CI currently still only lints the core /src files - not the example files.

Comment (mrdoob) [reply examples/jsm/exporters/GLTFExporter.js:2568]: Oh right right

File: examples/jsm/exporters/GLTFExporter.js (hunk 1 above)
Comment (mrdoob) [reply examples/jsm/exporters/GLTFExporter.js]: Fixed.

File: examples/jsm/exporters/GLTFExporter.js (hunk 2 above)
Comment (LeviPesin) [reply examples/jsm/exporters/GLTFExporter.js]: > @LeviPesin related to the latest CI changes maybe?

No -- I suggested in https://github.com/mrdoob/three.js/pull/25276 to expand the lint script to also check the linting of other directories but that idea didn't recieve much support.
Thinking about it again... maybe we can at least add a commit hook that will automatically run `npm run lint-fix` when user makes a commit via Git?

Comment (mrdoob) [reply examples/jsm/exporters/GLTFExporter.js]: That'd be a good start I think?

Comment (gkjohnson) [reply examples/jsm/exporters/GLTFExporter.js]: > Thinking about it again... maybe we can at least add a commit hook that will automatically run npm run lint-fix when user makes a commit via Git?

We have to be careful about how this is done. I commit individual lines when working through feature additions and every time I've used "auto linters" on commit it stages and commits the entire file including lines I didn't want included.

File: examples/jsm/exporters/GLTFExporter.js (hunk 1 above)
Comment (elalish) [reply examples/jsm/exporters/GLTFExporter.js]: Thanks! I thought it was too good to be true that the linter passed on my first try...

File: examples/jsm/exporters/GLTFExporter.js (hunk 2 above)
Comment (elalish) [reply examples/jsm/exporters/GLTFExporter.js]: I didn't even know about `npm run lint-fix`; that would probably save me a lot next time.

**Pull request merged**